java/arcs/core/storage/handle/HandleManager.kt

```diff
@@ -28,8 +32,8 @@ import kotlinx.coroutines.sync.withLock
  */
 interface ActivationFactoryFactory {
     fun dereferenceFactory(): EntityActivationFactory
-    fun singletonFactory(): SingletonActivationFactory<RawEntity>
-    fun setFactory(): SetActivationFactory<RawEntity>
+    fun <T : Referencable> singletonFactory(typeClass: KClass<T>): SingletonActivationFactory<T>
```

Comment:
Is this purely for type inference to capture T? If so, you might need an @Supress somewhere for unused parameter in the impl classes.

Reply:
Looks like we're okay.